### DISCUSSION TITLE
[Question] Is there a way to visit an element of array twice in ondemand API?

### DISCUSSION THREAD

**Discussion by zhanglistar** [simdjson/simdjson#2298] [Q&A]
Hi, all. I am working on using ondemand API to parse json. I have to visit an json data twice, first to get types and then elements, since they are split into two seperate classes, do we have this feature support?
Thanks a lot.

**Comment by lemire** [comment]
Please see the rewinding section of our documentation: 

https://github.com/simdjson/simdjson/blob/master/doc/basics.md#rewinding

**Comment by zhanglistar** [comment]
> Please see the rewinding section of our documentation:
> 
> https://github.com/simdjson/simdjson/blob/master/doc/basics.md#rewinding

@lemire Thanks  for your reply. I have read this documentation, and both `rewind` and `reset` can not satisfy my need directly(maybe `rewind` is OK, but I need refactor all code and flow). In fact, do we have or plan to do this API: get a position first, and then, you can restore it latter. For example, 
 auto cars_json = R"( [
    { "make": "Toyota", "model": "Camry",  "year": 2018, "tire_pressure": [ 40.1, 39.9, 37.7, 40.4 ] },
    { "make": "Kia",    "model": "Soul",   "year": 2012, "tire_pressure": [ 30.1, 31.0, 28.6, 28.7 ] },
    { "make": "Toyota", "model": "Tercel", "year": 1999, "tire_pressure": [ 29.8, 30.0, 30.2, 30.5 ] }
  ] )"_padded;
There is a position of `tire_pressure` in the second object, and I can still rewind to this place, and read the value again?
Thanks a lot.

**Comment by lemire** [comment]
> I can still rewind to this place, and read the value again?

If you need to access the same value multiple times, you should store it in your own data structure.
